app/src/main/java/at/tugraz/ist/guessingwords/data/service/WordService.kt

```diff
@@ -48,4 +48,46 @@ open class WordService(private val context : Context) {
             callback.whenReady(word)
         }
     }
+
+    /**
+     * Creates an In Memory Instance for merging Word Pools
+     * Populates it with local persistent Words
+     * @param callback
+     */
+    open fun createNewMultiplayerWordPool(callback: Callback<List<Long>>) {
+        thread {
+            val localWords = GWDatabase.getInstance(context).wordDao().getAll()
+            val dbTemp = GWDatabase.getInMemoryInstance(context)
+
+            val mergedIds = dbTemp.wordDao().mergeWordsIntoDB(localWords)
+            callback.whenReady(mergedIds)
+        }
+    }
+
+    /**
+     * Merges additional Words to the In Memory Instance
+     * Remove the duplicate Words before Merging
+     * @param merging
+     * @param callback
+     */
+    @OptIn(ExperimentalStdlibApi::class)
+    open fun mergeIntoDatabase(merging: List<Word>, callback: Callback<List<Long>>) {
+        thread {
+            val db = GWDatabase.getInMemoryInstance(context)
+            val localWords = GWDatabase.getInstance(context).wordDao().getAll()
```

Comment:
Do not use persistent words for checking duplicates but check against all current (in memory words) instead.
So use the inMemoryDB instead